language/documentation/book/src/abilities.md

```diff
@@ -0,0 +1,243 @@
+
+If a value has `drop`, all values contained inside of that value have `drop`.
+
+### `store`
+
+The `store` ability allows values of types with this ability to exist inside of a struct (resource) inside of global storage. *But* not necessarily as a top-level resource in global storage. This is the only ability that does not directly gate an operation. Instead it gates the existence inside of global storage when used in tandem with `key`.
+
+If a value has `store`, all values contained inside of that value have `store`
+
+### `key`
+
+The `key` ability allows the type to serve as a key for [global storage operations](./global-storage-operators.md). It gates all global storage operations, so in order for a type to be used with `move_to`, `borrow_global`, `move_from`, etc., the type must have the `key` ability. Note that the operations still must be used in the module where the `key` type is defined (in a sense, the operations are private to the defining module).
+
+If a value has `key`, all values contained inside of that value have `store`. This is the only ability with this sort of asymmetry.
+
+## Builtin Types
+
+Most primitive, builtin types have `copy`, `drop`, and `store` with the exception of `signer`, which just has `store`
+
+* `bool`, `u8`, `u64`, `u128`, and `address` all have `copy`, `drop`, and `store`.
+* `signer` has `drop`
+    * Cannot be copied and cannot be put into global storage
+* `vector<T>` may have `copy`, `drop`, and `store` depending on the abilities of `T`.
+    * See [Conditional Abilities and Generic Types](#conditional-abilities-and-generic-types) for more details.
+* Immutable references `&` and mutable references `&mut` both have `copy` and `drop`.
+    * This refers to copying and dropping the reference itself, not what they refer to.
+    * References cannot appear in global storage, hence they do not have `store`.
+
+None of the primitive types have `key`, meaning none of them can be used directly with the [global storage operations](./global-storage-operators.md).
+
+## Annotating Structs
+
+To declare that a `struct` has an ability, it is declared with `has <ability>` after the struct name but before the fields. For example:
+
+```move
+struct Ignorable has drop { f: u64 }
+struct Pair has copy, drop, store { x: u64, y: u64 }
+```
+
+In this case: `Ignorable` has the `drop` ability. `Pair` has `copy`, `drop`, and `store`.
+
+
+All of these abilities have strong guarantees over these gated operations. The operation can be performed on the value only if it has that ability; even if the value is deeply nested inside of some other collection!
+
+As such: when declaring a struct’s abilities, certain requirements are placed on the fields. All fields must satisfy these constraints. These rules are necessary so that structs satisfy the reachability rules for abilities given above. If a struct is declared with the ability...
```

Comment:
```suggestion
As such: when declaring a struct’s abilities, certain requirements are placed on the fields. All fields must satisfy these constraints. These rules are necessary so that structs satisfy the reachability rules for the abilities given above. If a struct is declared with the ability...
```